File: two-sum/twosum.py
class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        numbers = {}
        for i in range(len(nums)):
            if nums[i] not in numbers:
                numbers[nums[i]] = []
            numbers[nums[i]].append(i)
            complement = target-nums[i]
            if complement in numbers:
                if numbers[complement][0] == i:
                    continue
                return [numbers[complement][0],i]

File: two-sum/twosum.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* twoSum(int* nums, int numsSize, int target) {
    int i, j;
    for (i = 0; i < numsSize; i++){
        for (j = 1; j < numsSize; j++){
            if (i == j){
                continue;
            }
            if (nums[i]+nums[j] == target){
                int* ret = malloc(2*sizeof(int));
                ret[0] = i;
                ret[1] = j;
                return ret;
            }
        }
    }
    return NULL;
}